p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -268,9 +299,18 @@ public GetDataResponse buildGetDataResponse(
         Set<P2PDataStorage.ByteArray> excludedKeysAsByteArray =
                 P2PDataStorage.ByteArray.convertBytesSetToByteArraySet(getDataRequest.getExcludedKeys());
 
+        String specialKey = containsSpecialKey(excludedKeysAsByteArray);
+
+        Map<ByteArray, PersistableNetworkPayload> tmp;
+        if ("".equals(specialKey))
+            tmp = this.appendOnlyDataStoreService.getMap();
+        else {
+            tmp = this.appendOnlyDataStoreService.getMap("since " + specialKey);
+        }
```

Comment:
Is this the separation of old and new requests for data? More comments would be helpful for understanding.

Variable name `tmp` isn't really descriptive, better naming would help.

Reply:
> Is this the separation of old and new requests for data?

Correct. If there is a special key incoming we prefilter the data, if not, we do what we did before.